crates/masking/src/secret.rs

```diff
@@ -156,3 +156,10 @@ where
         SecretValue::default().into()
     }
 }
+
+// Required by base64-serde to serialize Secret
+impl<T> AsRef<[T]> for Secret<Vec<T>> {
+    fn as_ref(&self) -> &[T] {
+        self.peek().as_slice()
+    }
+}
```

Comment:
Make this implementation specific to `Vec<u8>` not for any `T`. (Update the comment as well: `// Required by base64-serde to serialize Secret<Vec<u8>>`.)